src/cmd/flux-job.c

```diff
@@ -632,13 +642,31 @@ int cmd_id (optparse_t *p, int argc, char **argv)
     return 0;
 }
 
+enum {
+    CONTEXT_FORMAT_TEXT = 0,
+    CONTEXT_FORMAT_JSON = 1,
+};
+
 struct eventlog_ctx {
     struct flux_kvs_eventlog *log;
     optparse_t *p;
     flux_jobid_t id;
+    int format;
 };
 
-void output_event (const char *event)
+int context_format (optparse_t *p)
+{
+    const char *type = optparse_get_str (p, "context-format", "text");
+    if (!strcasecmp (type, "text"))
+        return CONTEXT_FORMAT_TEXT;
+    else if (!strcasecmp (type, "json"))
+        return CONTEXT_FORMAT_JSON;
+    log_msg_exit ("invalid context-format type");
+    /* NEVER REACHED */
+    return -1;
+}
+
+void output_event (const char *event, int format)
```

Comment:
Suggestion: refactor to `output_event_json()` and `output_event_text()` and leave the format option checking to the main function for the subcommand?